### PR TITLE
Check numeric equality before using format

### LINKED ISSUE #134: Don't use format if ignore_attr = TRUE
``` r
waldo::compare(factor("a"), 1L, ignore_attr = TRUE)
#> `old`: "a"
#> `new`: "1"
```

<sup>Created on 2022-03-11 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### PR DESCRIPTION
**Pull request by hadley**
Fixes #134

### CODE REVIEW

File: tests/testthat/test-compare-value.R [hunk 1]
```diff
@@ -123,12 +123,18 @@ test_that("uses format method if available", {
...
-
 test_that("ignore_attr never uses format method", {
   expect_snapshot({
     compare(.POSIXct(1, "UTC"), .POSIXct(2, "UTC"), ignore_attr = TRUE)
   })
 })
+test_that("don't use format if numeric & within tolerance", {
+  dt <- as.POSIXct("2016-07-18 16:06:00", tz = "UTC")
+  expect_snapshot({
+    compare(dt, dt + 5)
```
Comment (MichaelChirico) [inline tests/testthat/test-compare-value.R:134]: that means timestamps 5 seconds apart are considered equal? seems strange to me

Comment (hadley) [reply tests/testthat/test-compare-value.R:134]: `compare()` never errors; see the snapshot for the actual results.

Comment (MichaelChirico) [reply tests/testthat/test-compare-value.R:134]: in what sense are dt and dt+5 within 1e-8 of each other?

Comment (hadley) [reply tests/testthat/test-compare-value.R:134]: The tolerance is relative, not absolute.

Comment (hadley) [reply tests/testthat/test-compare-value.R:134]: (This behaviour isn't new; it's just exposed by the change to use `format`. I'll tackle it in a separate issue.)

**Pull request merged**